tst/EnergyPlus/unit/OutputFiles.unit.cc

```diff
@@ -219,4 +219,244 @@ TEST_F(EnergyPlusFixture, OutputFiles_Expected_Formatting_Tests)
+        "  Yes,                     !- Output PerfLog",
+        "  Yes,                     !- Output SLN",
+        "  Yes,                     !- Output SCI",
+        "  Yes,                     !- Output WRL",
+        "  Yes,                     !- Output Screen",
+        "  Yes,                     !- Output ExtShd",
+        "  Yes;                     !- Output Tarcog",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    state->files.outputControl.getInput(*state);
+
+    state->dataGlobal->DisplayUnusedObjects = true;
+
+    state->dataInputProcessing->inputProcessor->reportOrphanRecordObjects(*state);
+
+    // It does not include "   **   ~~~   ** Object=OutputControl:Files=OutputControl:Files 1"
+    EXPECT_FALSE(match_err_stream("OutputControl:Files"));
+
+    std::string expected_error = delimited_string({
+
+        "   ** Warning ** The following lines are \"Unused Objects\".  These objects are in the input",
+        "   **   ~~~   **  file but are never obtained by the simulation and therefore are NOT used.",
+        "   **   ~~~   **  Only the first unused named object of an object class is shown.  Use Output:Diagnostics,DisplayAllWarnings; to see all.",
+        "   **   ~~~   **  See InputOutputReference document for more details.",
+        "   ************* Object=Building=Bldg",
+        "   **   ~~~   ** Object=GlobalGeometryRules",
+        "   **   ~~~   ** Object=Version",
+    });
+
+    compare_err_stream(expected_error);
+}
+
+TEST_F(EnergyPlusFixture, OutputControlFiles_GetInput)
+{
+    std::string const idf_objects_fmt = R"(
+OutputControl:Files,
+  {csv},              !- Output CSV
+  {mtr},              !- Output MTR
+  {eso},              !- Output ESO
+  {eio},              !- Output EIO
+  {tabular},          !- Output Tabular
+  {sqlite},           !- Output SQLite
+  {json},             !- Output JSON
+  {audit},            !- Output AUDIT
+  {zsz},              !- Output Zone Sizing
+  {ssz},              !- Output System Sizing
+  {dxf},              !- Output DXF
+  {bnd},              !- Output BND
+  {rdd},              !- Output RDD
+  {mdd},              !- Output MDD
+  {mtd},              !- Output MTD
+  {end},              !- Output END
+  {shd},              !- Output SHD
+  {dfs},              !- Output DFS
+  {glhe},             !- Output GLHE
+  {delightin},        !- Output DelightIn
+  {delighteldmp},     !- Output DelightELdmp
+  {delightdfdmp},     !- Output DelightDFdmp
+  {edd},              !- Output EDD
+  {dbg},              !- Output DBG
+  {perflog},          !- Output PerfLog
+  {sln},              !- Output SLN
+  {sci},              !- Output SCI
+  {wrl},              !- Output WRL
+  {screen},           !- Output Screen
+  {extshd},           !- Output ExtShd
+  {tarcog};           !- Output Tarcog
+  )";
+
+    auto boolToString = [](bool b) { return b ? "Yes" : "No"; };
+
+    for (int i = 0; i < 31; ++i) {
+        bool csv = (i == 0);
```

Comment:
Test every possible combination with just one as "True" to ensure everything is captured correctly in getInput